### PR TITLE
add basic telemetry implementation

### PR DESCRIPTION
**Pull request by d-ivashchuk**
Adds telemetry for OSS version

### CODE REVIEW

File: src/runner.ts [hunk 1]
```diff
@@ -36,17 +39,23 @@ export const runner = async () => {
...
 
     log('Creating shot folders');
+    const createShotsStart = process.hrtime();
     createShotsFolders();
 
     log('Creating shots');
     const shotItems = await createShots();
 
+    const createShotsStop = process.hrtime(createShotsStart);
+    log(`Creating shots took:${parseHrtimeToSeconds(createShotsStop)}`);
```
Comment (chriskalmar) [inline src/runner.ts:49]:
```suggestion
    log(`Creating shots took ${parseHrtimeToSeconds(createShotsStop)} seconds`);
```

File: src/runner.ts [hunk 2]
```diff
@@ -66,15 +75,51 @@ export const runner = async () => {
       process.exit(1);
```
Comment (chriskalmar) [inline src/runner.ts:75]: also here we would need something

Comment (d-ivashchuk) [reply src/runner.ts:80]: 7997a9e705381d15529ac7dc9a910e743e727e60 added the telemetry send in cases where we exit process earlier

File: src/runner.ts [hunk 3]
```diff
@@ -36,17 +39,24 @@ export const runner = async () => {
...
 
     log('Creating shots');
     const shotItems = await createShots();
 
+    const createShotsStop = process.hrtime(createShotsStart);
+    log(`Creating shots took ${parseHrtimeToSeconds(createShotsStop)} seconds`);
+
     if (config.generateOnly && shotItems.length === 0) {
       log(`Exiting process with nothing to compare.`);
+      log(`Process took ${parseHrtimeToSeconds(createShotsStop)} seconds`);
```
Comment (chriskalmar) [inline src/runner.ts:53]: How about having a `shutdown` function that will include the log message about how long the process took and send the telemetry data. This way we would have a consistent behaviour on every `process.exit()`

Comment (d-ivashchuk) [reply src/runner.ts:53]: 👌 sounds good - will do this

File: src/runner.ts [hunk 4]
```diff
@@ -36,17 +39,23 @@ export const runner = async () => {
...
 
     log('Creating shot folders');
+    const createShotsStart = process.hrtime();
     createShotsFolders();
 
     log('Creating shots');
     const shotItems = await createShots();
 
+    const createShotsStop = process.hrtime(createShotsStart);
+    log(`Creating shots took ${parseHrtimeToSeconds(createShotsStop)} seconds`);
```
[the remark this answers is not in the document]
Comment (d-ivashchuk) [reply src/runner.ts:49]: I don't think that `exitProcess` should be concerned with that much external info. removed this line though - seems redundant

[the remark this answers is not in the document]
Comment (d-ivashchuk) [reply src/runner.ts:49]: [b3b5683](https://github.com/lost-pixel/lost-pixel/pull/172/commits/b3b56835863b218044121cb98e49b81c7c2c5679)

**Review by chriskalmar — APPROVED** [review]

**Pull request merged**